components/service/firefox-accounts/src/main/java/mozilla/components/service/fxa/manager/FxaAccountManager.kt

```diff
@@ -1008,7 +1008,12 @@ open class FxaAccountManager(
         return if (deviceConfig.secureStateAtRest) {
             SecureAbove22AccountStorage(context, crashReporter)
         } else {
-            SharedPrefAccountStorage(context, crashReporter)
+            // Don't migrate here so we don't hit the KeyStore path if the secureStateAtRest flag is false
```

Comment:
This change is interesting - it removes our ability to flip back and forth between the two storage modes, changing the public API in a subtle way.

Before, application could switch `secureStateAtRest` in between versions without any problems, in theory - state would be migrated between storage layers.

After this change, if the application has secureStateAtRest=true, but then switches it to `false`, FxA state will be lost. So flipping this flag and releasing, every single user will get logged out. That's a pretty big foot gun!

I'm not against doing this temporarily while we learn more about scale of they keystore problems, but we should really put up a big, fat warning sign both in Fenix where this flag is set, and in `DeviceConfig` as well.